### PR TITLE
Display "Generate RTL stylesheets" section

### PR DESCRIPTION
**Pull request by ramtin2025**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | "rtl"
| Description?  | Display "Generate RTL stylesheets" section if there is at least one RTL language
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | See the "themes and logo" section on BO

This PR contains:
- Modified AdminThemesController.php and options.tpl

### CODE REVIEW

**Comment by AlexEven** [comment]
Hello @ramtin2025 

Thanks for this PR
A bit late, but here are some more specifications regarding this topic.
https://trello.com/c/ibl7fzct/82-add-option-to-generate-css-file-for-theme

Thank you!

**Comment by maximebiloe** [comment]
Hello @ramtin2025,

Did you check @AlexEven's mockups on the Trello card?
Is it ok for you?

Also, you still have commits that do not respect our standards. Can you update them, please?

Regards

**Comment by mehrshadz** [comment]
Hello @maximebiloe 

@ramtin2025 is on a vacation and he'll be back soon. Then we'll continue our PRs and update the non-standard ones.

Thanks

**Comment by mehrshadz** [comment]
@maximebiloe @AlexEven 
We've updated the PR. Please check it

cheers

File: controllers/admin/AdminThemesController.php [hunk 1]
```diff
@@ -325,6 +325,13 @@ public function postProcess()
                 $this->theme_manager->uninstall(Tools::getValue('theme_name'));
                 $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
             }
+        } elseif (Tools::isSubmit('submitGenerateRTL') && Tools::getValue('IPS_GENERATE_RTL')) {
+            Language::installRtlStylesheets(false, true, Tools::getValue('IPS_THEMES_LIST'));
+            $this->confirmations[] = $this->trans(
+                'Your RTL stylesheets has been generated successfully',
```
Comment (xBorderie) [inline controllers/admin/AdminThemesController.php]: If one stylesheet: `Your RTL stylesheet has been...`.
If several stylesheets: `Your RTL stylesheets have been...`.

File: controllers/admin/AdminThemesController.php [hunk 2]
```diff
@@ -533,6 +540,38 @@ public function renderOptions()
...
+        if (in_array("1", array_column($this->_languages, 'is_rtl'))) {
+            $themes_list = array();
+            $allThemes = $this->theme_repository->getList();
+            foreach ($allThemes as $theme) {
+                $themeName = $theme->getName();
+                $themes_list[] = array('theme' => $themeName, 'name' => $themeName);
+            }
+            $this->fields_options['RTL'] = array(
+                'title' => sprintf($this->trans('Adaptation to Right-to-Left languages', array(), 'Admin.Design.Feature'), $this->context->shop->name),
+                'description' => $this->trans('Be careful! Please check your theme in an RTL language before generating the RTL stylesheet: your theme could be already adapted to RTL.\nOnce you click on "Adapt to RTL", the stylesheet created may delete any files you could have previously in your theme to adapt to RTL.', array(), 'Admin.Design.Help'),
```
Comment (xBorderie) [inline controllers/admin/AdminThemesController.php]: I think it should be `...delete any file...` (singular).

I find the last sentence a bit complex to read. I suggest an alternative: `Once you click on "Adapt to RTL", any RTL-specific file that you might have added to your theme might be deleted by the created stylesheet`. What do you think @AlexEven ?
Also, does this mean that the new stylesheet needs non-CSS files to be deleted? Or should that be explained in more details?

File: controllers/admin/AdminThemesController.php [hunk 3]
```diff
@@ -325,6 +325,13 @@ public function postProcess()
                 $this->theme_manager->uninstall(Tools::getValue('theme_name'));
                 $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
             }
+        } elseif (Tools::isSubmit('submitGenerateRTL') && Tools::getValue('IPS_GENERATE_RTL')) {
```
Comment (maximebiloe) [inline controllers/admin/AdminThemesController.php]: Why `IPS_` as a prefix?
I would prefer `PS_` only 😉

File: controllers/admin/AdminThemesController.php (hunk 2 above)
Comment (AlexEven) [reply controllers/admin/AdminThemesController.php]: Thanks @xBorderie for the review. Yes, we can go with your suggestion! 👍

**Comment by AlexEven** [comment]
Thank you guys for the update! :)

File: controllers/admin/AdminThemesController.php [hunk 4]
```diff
@@ -533,6 +540,38 @@ public function renderOptions()
...
+		
+        if (in_array("1", array_column($this->_languages, 'is_rtl'))) {
+            $themes_list = array();
+            $allThemes = $this->theme_repository->getList();
+            foreach ($allThemes as $theme) {
+                $themeName = $theme->getName();
+                $themes_list[] = array('theme' => $themeName, 'name' => $themeName);
+            }
+            $this->fields_options['RTL'] = array(
+                'title' => sprintf($this->trans('Adaptation to Right-to-Left languages', array(), 'Admin.Design.Feature'), $this->context->shop->name),
```
Comment (maximebiloe) [inline controllers/admin/AdminThemesController.php]: The `sprintf()` is useless here I think 🤔

File: controllers/admin/AdminThemesController.php [hunk 5]
```diff
@@ -533,6 +540,38 @@ public function renderOptions()
...
+                'description' => $this->trans('Be careful! Please check your theme in an RTL language before generating the RTL stylesheet: your theme could be already adapted to RTL.\nOnce you click on "Adapt to RTL", the stylesheet created may delete any files you could have previously in your theme to adapt to RTL.', array(), 'Admin.Design.Help'),
+                'fields' => array(
+                    'IPS_THEMES_LIST' => array(
```
Comment (maximebiloe) [inline controllers/admin/AdminThemesController.php]: Same here about the prefix

File: controllers/admin/AdminThemesController.php [hunk 6]
```diff
@@ -533,6 +540,38 @@ public function renderOptions()
...
+                        'title' => $this->trans('Theme to adapt', array(), 'Admin.Design.Feature'),
+                        'type' => 'select',
+                        'identifier' => 'theme',
+                        'list' => $themes_list,
+                    ),
+                    'IPS_GENERATE_RTL' => array(
```
Comment (maximebiloe) [inline controllers/admin/AdminThemesController.php]: Same here about the prefix

**Review by maximebiloe — CHANGES_REQUESTED** [review]

**Review by maximebiloe — DISMISSED** [review]

**Comment by maximebiloe** [comment]
Hello @ramtin2025,

We had some issues with Travis for our tests. Would you mind rebasing your PR, please?

Thank you

**Comment by Danoosh** [comment]
@maximebiloe Please check this again.

**Comment by AlexEven** [comment]
Thank you @Danoosh!
We'll review this quickly to merge it.

**Pull request merged**